File: typescript/src/shared/price.ts
/**
 * Parse a decimal string to a number.
 */
export function parseDecimal(value: string): number {
  const parsed = Number.parseFloat(value);
  if (Number.isNaN(parsed)) {
    throw new Error(`Invalid decimal string: ${value}`);
  }
  return parsed;
}

/**
 * Format a number to a fixed decimal string.
 */
export function formatDecimal(value: number, precision: number): string {
  return value.toFixed(precision);
}

/**
 * Whether a decimal string represents numeric zero.
 */
export function isZero(value: string): boolean {
  const parsed = Number.parseFloat(value);
  return !Number.isNaN(parsed) && parsed === 0;
}

File: python/src/shared/price.py
"""Price utilities used across the Lightcone SDK."""

from decimal import Decimal


def parse_decimal(s: str) -> Decimal:
    """Parse a decimal string to ``Decimal``."""
    return Decimal(s)


def format_decimal(value: Decimal | float | str, precision: int = 6) -> str:
    """Format a value as a decimal string with the specified precision.

    The implementation is ``Decimal``-first to avoid float-driven rounding
    surprises in request payloads and typed domain conversions.
    """
    decimal_value = value if isinstance(value, Decimal) else Decimal(str(value))
    quantized = decimal_value.quantize(Decimal(1).scaleb(-precision))
    return format(quantized, "f")


def is_zero(value: str) -> bool:
    """Check if a decimal string represents zero."""
    try:
        return Decimal(value) == 0
    except Exception:
        return False
